conan/tools/cmake/toolchain/blocks.py

```diff
@@ -815,6 +815,11 @@ def _get_generic_system_name(self):
                 return cmake_system_name_map.get(os_host, os_host)
 
     def _is_apple_cross_building(self):
+
+        if _is_universal_arch(self._conanfile.settings.get_safe("arch"),
+                              self._conanfile.settings.possible_values()):
```

Comment:
This looks wrong, the ``possible_values`` will return {"os": ..., "arch": ....}, but then internally you are iterating it for ``arch`` values only, without getting the ``arch`` values. Not sure what I am not seeing here, the tests are passing, but at least the unittest is passing the wrong values.

Reply:
Yes, sorry, I'll fix that